src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java

```diff
@@ -1557,6 +1561,26 @@ private static boolean isCreditNote(
                 .equals(paymentDetail.getFinancialDocumentTypeCode());
     }
 
+    /*
+     * CU Customization: Added new method for deriving the document/invoice number.
+     */
+    private static String determineReferredDocumentNumber(
+            final PaymentDetail paymentDetail
+    ) {
+        if (StringUtils.equalsAny(paymentDetail.getFinancialDocumentTypeCode(),
+                DisbursementVoucherConstants.DOCUMENT_TYPE_CHECKACH,
+                DisbursementVoucherConstants.DOCUMENT_TYPE_CODE)) {
+            if (StringUtils.isNotBlank(paymentDetail.getInvoiceNbr()) && paymentDetail.getInvoiceDate() != null) {
+                return paymentDetail.getInvoiceNbr();
+            } else {
+                return CuDisbursementVoucherConstants.DV_EXTRACT_EDOC_NUMBER_PREFIX_IDENTIFIER
```

Comment:
what is the source of this payment?

Reply:
As per the `StringUtils.isAny()` comparison a few lines above, the payment is sourced from a DV if execution reaches this marked line. As far as I can tell, DV-sourced payments appear to be the only ones that can potentially have a blank invoice number.

Please let me know if you need me to make the is-DV check clearer.

Reply:
Are these DV's created in a different manner? Or is just a condition that can appear on any DV? Thank you.

Reply:
DVs have fields in the "Payment Information" section for specifying the Invoice Number and Invoice Date, but they appear to be optional fields. As a result, there are cases where a DV document does not have an Invoice Number.